t/t1092-sparse-checkout-compatibility.sh

```diff
@@ -387,6 +387,45 @@ test_expect_success 'diff --staged' '
 	test_all_match git diff --staged
 '
 
+test_expect_success 'diff partially-staged' '
+	init_repos &&
+
+	write_script add-file-in-cone <<-\EOF &&
+	git sparse-checkout set deep
+	echo >deep/testfile
+	git add deep/testfile
+	echo a new line >>deep/testfile
+	EOF
+	run_on_all ../add-file-in-cone &&
+
+	test_all_match git diff &&
+	test_all_match git diff --staged &&
+
+	write_script add-file-outside-cone <<-\EOF &&
+	git reset --hard
+	mkdir newdirectory
+	echo >newdirectory/testfile
+	git sparse-checkout set newdirectory
+	git add newdirectory/testfile
+	echo a new line >>newdirectory/testfile
+	git sparse-checkout set
+	EOF
```

Comment:
Since this script is used only once, it probably doesn't need to be a script at all, and the commands could be written out individually:

```
	test_all_match git reset --hard &&
	run_on_all mkdir newdirectory &&
	run_on_all echo >newdirectory/testfile &&
	test_all_match git sparse-checkout set newdirectory &&
	test_all_match git add newdirectory/testfile &&
	run_on_all echo a new line >>newdirectory/testfile &&
	test_all_match git sparse-checkout set &&
```

This allows you to make extra checks (`test_all_match`) while also cutting out the step of writing the script. The same thing can be done for `add-file-in-cone` and `merge-conflict-outside-cone`.

Reply:
Excellent suggestion, thank you!

Reply:
As a heads up, I had to add `'` around all the `echo` commands to make this work, but aside from that I just applied the pattern you specified here!